stage4.py

```diff
@@ -0,0 +1,48 @@
+from card import Card
+import random
+
+
+deck = list()
+cards_in_deck = 0
+is_card_identical = 0
+while cards_in_deck < 40:
+    pulled_card = Card()
+    for card in deck:
+        if card == pulled_card:
+            is_card_identical = 1
+            break
+    if is_card_identical == 1:
+        is_card_identical = 0
+        continue
+    else:
+        cards_in_deck += 1
+        deck.append(pulled_card)
+
+random.shuffle(deck)
+
+num_players = 0
```

Comment:
shortcuts create confusion. 
I could understand it like מספר השחקן 

number_of_players 
or 
players_count

Reply:
Hint: you have a shortcut in pycharm to rename a variable, search for it.